### ISSUE TITLE
🚧 Dev ➾ Document Instructions for Using Beacon Wallet on a Sandbox

### ISSUE THREAD

**Issue by hu3man** [ecadlabs/taqueria#1172]
@### 🔗 Epic|Feature Link

https://ecadlabs.productboard.com/feature-board/5219572-taqueria-features/features/14699515

### 🚥 Definition of Done

- [x] 🎯 Planning and Design
- [x] 🏎️ Dev Implementation
- [ ] 🏟️ Peer / User Reviewed
- [ ] 🏆 Validated and Accepted

### ✅ To Do

- [ ] Slice out new feature ticket from suggestion @hu3man 
- [ ] Add tests @mweichert 
- [ ] Add docs and reference to demo dapp to the website 

### 🔭 Scope

🏎️ In Scope: 
 - [x] Writing the series of steps required to connect Beacon to a local Flextesa instance for use in dApp development
 - [x] Code snippets or a concrete example of how to make this work

 🙅‍♂️ Out of Scope: 
 - [ ] Planning for any wallet integration features
 - [ ] ANY development work on the Taqueria repo
 - [ ] 
 
 

### 🧰 Work Description

Several team members have been unable to get Beacon wallet working with Flextesa in a dApp. In order to understand the friction points in the process and unblock team members, we need a clear and concise understanding of how to make Beacon work with Flextesa

If there is a blocker or issue that prevents this from being possible, clearly identify the issue and any possible solutions (of any complexity)

### ⚖️ Acceptance Criteria

✨ Features: 
- [x] None! 
 
 📚 Documentation: 
- [ ] Step by step instructions for configuring and connecting to Beacon in a Flextesa environment
- [ ] 
 

### 🐘 Sizing

0.5

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mweichert** [comment]
I've created some modifications in #1296 to ensure that users can interact with local sandboxes without CORS issues. No proxying services like Caddy are required, nor do we require any specifc interaction with matrix nodes.

> Writing the series of steps required to connect Beacon to a local Flextesa instance for use in dApp development

Here are the summary of required steps necessary to connect your dapp to a local sandbox:

1) Start the sandbox
2) In your dapp:
```typescript
// Import your taqueria config.json file so that we can dynamically
// determine the the sandbox label & RPC url
import taqConfig from "taq/config.json"

// Use Taquito and the Beacon SDK to add the sandbox network to your wallet
import { TezosToolkit } from "@taquito/taquito";
import { BeaconWallet } from "@taquito/beacon-wallet"

const Tezos = new TezosToolkit(taqConfig.sandbox.local.rpcUrl);

const wallet = new BeaconWallet({
  name: "Beacon/Sandbox Test"
});

Tezos.setWalletProvider(wallet)

wallet.requestPermissions({
      network: {
        type: 'custom' as NetworkType,
        name: taqConfig.sandbox.local.label,
        rpcUrl: taqConfig.sandbox.local.rpcUrl
      }
```
3) Use `taq transfer` to transfer funds from Bob to your wallet address.

You can retrieve your public key hash using
```typescript
const current = await wallet.client.getActiveAccount()
console.log(current)
```

> Code snippets or a concrete example of how to make this work

I created a demo dapp to showcase how you can connect to a sandbox:
https://github.com/ecadlabs/taqueria-beacon-sandbox

**Comment by mweichert** [comment]
## Further Insight

From a developer workflow standpoint, I think we're going about things the wrong way with regards to how we generate sandbox test accounts.

Some points to note:
- Most developers will write dApps for users that will interact with their dApp using a Beacon Wallet implementation, such as Temple.
- It should probably be anti-pattern to use the InMemorySigner, whether in development or production.  It might be ok for very rapid prototyping, but behind that, developers should really have tools that make it dead simple to get started with a Beacon Wallet implementation

Taking the above notes into consideration, Taqueria should seize the opportunity. 

### Understanding a dApp developer's workflow

With the above notes in mind, we should be encouraging developers to install a Beacon Wallet implementation, such as Temple and AirGap. Given how prominent software wallets are in the cryptoverse, I think there's a high likelihood that developers will already have a wallet installed and that they're familiar with. As such, "getting a wallet" should be their first step.

Now that a developer has a wallet, they can create accounts from within the wallet. For instance, the developer might have an account that represents their own identity, and then create some test accounts, such as Bob, Alice, and Bill.

Their next step should be to start a sandbox. This is where I think we need to do things differently. When you run `taq start sandbox local`, instead of generating accounts right away, we should ask to connect to their Beacon Wallet. Once connected, we can call the getAccounts() method using the Beacon SDK to present them a list of accounts. 

We could then display something like this:
```shell
Found accounts: Mike, Bob, Alice, and Bill

Select an option below:
1) Fund all accounts with the same amount of Tz
2) Fund each account with a specific amount of Tz

Selection (1 or 2, default: 1): [  ]
```

If a developer selects `1` then they're asked for an amount. Otherwise they're prompted to enter an amount for each account.

The data collected above becomes the default template for any sandbox.

**Comment by mweichert** [comment]
Continuing on the thought above...

Once a default funding template has been created for a sandbox, that replaces our means of declarative accounts. So the next time you start a sandbox, such as `taq start sandbox localghost`, it would use the template to instantiate the sandbox-specific accounts.
